src/main/java/se/sundsvall/supportmanagement/service/ErrandAttachmentService.java

```diff
@@ -18,67 +20,65 @@
 import se.sundsvall.supportmanagement.api.model.attachment.ErrandAttachmentHeader;
 import se.sundsvall.supportmanagement.integration.db.AttachmentRepository;
 import se.sundsvall.supportmanagement.integration.db.ErrandsRepository;
-import se.sundsvall.supportmanagement.integration.db.model.AttachmentEntity;
 import se.sundsvall.supportmanagement.integration.db.model.ErrandEntity;
+import se.sundsvall.supportmanagement.service.mapper.ErrandAttachmentMapper;
 
 @Service
+@Transactional
 public class ErrandAttachmentService {
 
 	private static final String ERRAND_ENTITY_NOT_FOUND = "An errand with id '%s' could not be found in namespace '%s' for municipality with id '%s'";
-	private static final String ATTACHMENT_ENTITY_NOT_FOUND = "An attachment with id '%s' could not be found";
+	private static final String ATTACHMENT_ENTITY_NOT_FOUND = "An attachment with id '%s' could not be found on errand with id '%s'";
 	private static final String ATTACHMENT_ENTITY_NOT_CREATED = "Attachment could not be created";
-	private static final String ATTACHMENT_ENTITY_DO_NOT_BELONG_TO_ERRAND = "Attachment with id '%s' was not found for errand with id '%s'";
 
 	@Autowired
 	private AttachmentRepository attachmentRepository;
 
 	@Autowired
 	private ErrandsRepository errandsRepository;
 
-	public String createErrandAttachment(String namespace, String municipalityId, String errandId, ErrandAttachment errandAttachment) {
-		final var errandEntity = errandsRepository.findById(errandId)
-			.filter(entity -> Objects.equals(namespace, entity.getNamespace()))
-			.filter(entity -> Objects.equals(municipalityId, entity.getMunicipalityId()))
-			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, String.format(ERRAND_ENTITY_NOT_FOUND, errandId, namespace, municipalityId)));
+	@Autowired
+	private RevisionService revisionService;
 
+	public String createErrandAttachment(String namespace, String municipalityId, String errandId, ErrandAttachment errandAttachment) {
+		final var errandEntity = getErrand(errandId, namespace, municipalityId);
 		final var attachmentEntity = ofNullable(toAttachmentEntity(errandEntity, errandAttachment)).orElseThrow(() -> Problem.valueOf(BAD_GATEWAY, ATTACHMENT_ENTITY_NOT_CREATED));
-		return attachmentRepository.save(attachmentEntity).getId();
+		final var attachmentId = attachmentRepository.save(attachmentEntity).getId();
```

Comment:
Why are we using attachmentRepository here? Why not add the element on the collection (in errandEntity) and save?

This will override prePersist/preUpdate on the errandEntity, right? (i.e. supress the timestamp-setting)

Reply:
Yes, I saw that too. Removed attachmentRepo and refactored to use errandRepo instead.